File: shipments/create-shipment.js
const Easypost = require('@easypost/api');
const dotenv = require('dotenv');

dotenv.config({ path: '/Users/jhammond/git/easypost/easypost-tools/.env' });
const api = new Easypost(process.env.EASYPOST_PROD_API_KEY);

var hrstart = process.hrtime(); // TODO: Fix benchmark timing

/* Either objects or ids can be passed in. If the object does
 * not have an id, it will be created. */

/*
const toAddress = new api.Address({ 'id': 'adr_d739b50e314e48e09c3aab05b4179893' });
const fromAddress = new api.Address({ 'id': 'adr_d739b50e314e48e09c3aab05b4179893' });
const parcel = new api.Parcel({ 'id': 'prcl_76800e31b2c74a408c1132d19f19014a' });
// const customsInfo = new api.CustomsInfo({ ... });

const shipment = new api.Shipment({
  to_address: toAddress,
  from_address: fromAddress,
  parcel: parcel,
  //options: {
  //  alcohol: true,
  //},
  //customs_info: customsInfo
  carrier_accounts: [process.env.DPD],
});
*/

const shipment = new api.Shipment({
  to_address: {
    name: 'Dr. Steve Brule',
    street1: '179 N Harbor Dr',
    city: 'Redondo Beach',
    state: 'CA',
    zip: '90277',
    country: 'US',
    phone: '3331114444',
    email: 'dr_steve_brule@gmail.com'
  },
  from_address: {
    name: 'EasyPost',
    street1: '417 Montgomery Street',
    street2: '5th Floor',
    city: 'San Francisco',
    state: 'CA',
    zip: '94104',
    country: 'US',
    phone: '3331114444',
    email: 'support@easypost.com'
  },
  parcel: {
    length: 20.2,
    width: 10.9,
    height: 5,
    weight: 65.9
  },
  //options: {
  //},
  //reference: "PO:40657  Inv:20-0160-1\rPO:40657  Inv:20-0160-1",
  carrier_accounts: [process.env.DHL_EXPRESS],
});


shipment.save().then(console.log);
var hrend = process.hrtime(hrstart); // TODO: Fix benchmark timing
console.log("Execution time (hr): ", hrend[0], hrend[1]/1000000); // TODO: Fix benchmark timing

File: shipments/c#/createShipment.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using EasyPost;
using Newtonsoft.Json;

namespace EasyPost_shipment_test
{
    class createShipment
    {
        static void Main(string[] args)
        {
            EasyPost.ClientManager.SetCurrent(""); // TODO: .env variable here

            /*
            Address toAddress = Address.Create(toAddress);
            Address fromAddress = Address.Create(fromAddress);
            Parcel parcel = Parcel.Create(parcel);
            CustomsInfo customs = CustomsInfo.Create(customsInfo);

            Shipment shipment = new Shipment() {
            to_address = toAddress,
            from_address = fromAddress,
            parcel = parcel,
            customs_info = customs
            };

            shipment.Create();
            */

            // OR create in one call

            Dictionary<string, object> fromAddress = new Dictionary<string, object>() {
            { "name", "Dr. Steve Brule" },
            { "street1", "417 Montgomery Street" },
            { "street2", "5th Floor" },
            { "city", "San Francisco" },
            { "state", "CA" },
            { "country", "US" },
            { "zip", "94104" }
            };

            Dictionary<string, object> toAddress = new Dictionary<string, object>() {
            { "company", "EasyPost" },
            { "street1", "417 Montgomery Street" },
            { "street2", "Floor 5" },
            { "city", "San Francisco" },
            { "state", "CA" },
            { "country", "US" },
            { "zip", "94104" }
            };

            Shipment shipment = Shipment.Create(new Dictionary<string, object>() {
            { "parcel", new Dictionary<string, object>() {
                { "length", 8 }, { "width", 6 }, { "height", 5 }, { "weight", 10 }
            } },
            { "to_address", toAddress },
            { "from_address", fromAddress },
            //{ "reference", "ShipmentRef" }
            });

            Console.WriteLine(JsonConvert.SerializeObject(shipment, Formatting.Indented));
        }
    }
}
